Samples/src/OracleSynthesis/OracleSynthesis.qs

```diff
@@ -0,0 +1,298 @@
+        set res[i] = (PositiveIntFromBoolArr(current), j);
+        if (j < n) {
+            set current[j] = not current[j];
+        }
+      }
+
+      return res;
+    }
+
+    /// # Summary
+    /// Implements oracle circuit for function
+    ///
+    /// # Input
+    /// ## func
+    /// Oracle function in truth table representation
+    /// ## controls
+    /// Control qubits
+    /// ## target
+    /// Target qubit
+    operation Oracle(func : Bool[], controls : Qubit[], target : Qubit) : Unit {
+        let vars = Length(controls);
+        let table = Encode(func);
+        let spectrum = Extend(FastHadamardTransform(table));
+
+        let qubits = controls + [target];
+
+        HY(target);
+
+        for (i in 0..vars) {
+            let start = 1 <<< i;
+            let code = GrayCode(i);
+            for (j in 0..Length(code) - 1) {
+                let (offset, ctrl) = code[j];
+                RFrac(PauliZ, -spectrum[start + offset], vars + 2, qubits[i]);
+                if (i != 0) {
+                    CNOT(qubits[ctrl], qubits[i]);
+                }
+            }
+        }
+
+        H(target);
+    }
+
+    /// # Summary
+    /// Implements oracle circuit for function, assuming that target qubit
+    /// is initialized 0.  The adjoint operation assumes that the target
+    /// qubit will be released to 0.
+    ///
+    /// # Input
+    /// ## func
+    /// Oracle function in truth table representation
+    /// ## controls
+    /// Control qubits
+    /// ## target
+    /// Target qubit
+    operation OracleAncilla(func : Bool[], controls : Qubit[], target : Qubit) : Unit {
+        body (...) {
+            let vars = Length(controls);
+            let table = Encode(func);
+            let spectrum = Extend(FastHadamardTransform(table));
+
+            HY(target);
+
+            let code = GrayCode(vars);
+            for (j in 0..Length(code) - 1) {
+                let (offset, ctrl) = code[j];
+                RFrac(PauliZ, spectrum[offset], vars + 2, target);
+                CNOT(controls[ctrl], target);
+            }
+
+            H(target);
+        }
+        adjoint (...) {
+            let vars = Length(controls);
+            let table = Encode(func);
+            let spectrum = Extend(FastHadamardTransform(table));
+
+            H(target);
+            if (IsResultOne(M(target))) {
+                for (i in 0..vars - 1) {
+                    let start = 1 <<< i;
+                    let code = GrayCode(i);
+                    for (j in 0..Length(code) - 1) {
+                        let (offset, ctrl) = code[j];
+                        RFrac(PauliZ, -spectrum[start + offset], vars + 1, controls[i]);
+                        if (i != 0) {
+                            CNOT(controls[ctrl], controls[i]);
+                        }
+                    }
+                }
+                Reset(target);
+            }
+        }
+    }
+
+    /// # Summary
+    /// Operation to test Oracle operation
```

Comment:
perhaps add "Test" to the end of operation names in this section of the code? (so it is picked up by the test discovery tool).

Reply:
I'm not sure if the test discovery tool currently searches this project. Perhaps the easiest way is to add a ProjectReference to the UnitTesting sample, and to port tests there.

Reply:
I think that these are not really tests, but my entry routines from my C# driver.